Adamant/Modules/Settings/Notifications/NotificationsView.swift

```diff
@@ -0,0 +1,234 @@
+//
+//  NotificationsView.swift
+//  Adamant
+//
+//  Created by Yana Silosieva on 05.08.2024.
+//  Copyright © 2024 Adamant. All rights reserved.
+//
+
+import SwiftUI
+import CommonKit
+
+struct NotificationsView: View {
+    @StateObject var viewModel: NotificationsViewModel
+    let screensFactory: ScreensFactory
+    
+    init(
+        viewModel: @escaping () -> NotificationsViewModel,
+        screensFactory: ScreensFactory
+    ) {
+        _viewModel = .init(wrappedValue: viewModel())
+        self.screensFactory = screensFactory
+    }
+    
+    var body: some View {
+        GeometryReader { geometry in
+            Form {
+                notificationsSection()
+                messageSoundSection()
+                messageReactionsSection()
+                inAppNotificationsSection()
+                settingsSection()
+                moreDetailsSection()
+            }
+            .withoutListBackground()
+            .background(Color(.adamant.secondBackgroundColor))
+            .navigationBarTitleDisplayMode(.inline)
+            .toolbar {
+                ToolbarItem(placement: .principal) {
+                    toolbar(maxWidth: geometry.size.width)
+                }
+            }
+            .sheet(isPresented: $viewModel.presentSoundsPicker, content: {
+                NavigationView(content: {
+                    screensFactory.makeNotificationSounds(target: .baseMessage)
+                })
+            })
+            .sheet(isPresented: $viewModel.presentReactionSoundsPicker, content: {
+                NavigationView(content: {
+                    screensFactory.makeNotificationSounds(target: .reaction)
+                })
+            })
+            .fullScreenCover(isPresented: $viewModel.openSafariURL) {
+                SafariWebView(url: viewModel.safariURL).ignoresSafeArea()
+            }
+        }
+    }
+}
+
+private extension NotificationsView {
+    func toolbar(maxWidth: CGFloat) -> some View {
+        HStack {
+            Text(viewModel.notificationsTitle)
+                .font(.headline)
+                .minimumScaleFactor(0.7)
+                .lineLimit(1)
+        }
+        .frame(maxWidth: maxWidth - toolbarSpace, alignment: .center)
+    }
+    
+    func notificationsSection() -> some View {
+        Section {
+            Button(action: {
+                viewModel.showAlert()
+            }, label: {
```

Comment:
We have to reuse code. And this kind of button has already been used in the ContributeView. Check it out. Maybe you can modify it a bit and use the reusable solution

Reply:
I don't exactly understand what you mean. Let's keep it for the future

Reply:
responded in dm